io/disk/port_bounce.py

```diff
@@ -191,23 +199,49 @@ def verify_port_enable(self, test_ports):
                 msg = "Port %s is failed to enable" % port
                 self.failure_list[port] = msg
 
-    def verify_port_toggle_host(self, status):
-        """
-        Verifies port enable/disable status change in host
-        side for corresponding fc adapter
-        """
-        for bus_id in self.pci_bus_addrs:
-            pci_class = pci.get_pci_class_name(bus_id)
-            intf = pci.get_interfaces_in_pci_address(bus_id, pci_class)[-1]
-            state = genio.read_file("/sys/class/fc_host/%s/port_state"
-                                    % intf).rstrip("\n")
-            self.log.info("Host bus: %s, state: %s", bus_id, state)
-            if state == status:
-                self.log.info("bus: %s, port status %s got reflected",
-                              bus_id, state)
-            else:
-                self.fail("port state not changed in host expected state: %s, \
-                          actual_state: %s", status, state)
+    def mpath_fail_check(self, test_ports):
+        '''
+        checking mpath disk status after disabling the switch port
+        '''
+        total_mpaths = 0
+        count = 0
+        wwids = multipath.get_multipath_wwids()
+        for wwid in wwids:
+            if wwid not in process.system_output('multipath -ll',
+                                                 ignore_status=True,
+                                                 shell=True):
+                continue
+            mpaths = multipath.get_paths(wwid)
+            total_mpaths += len(mpaths)
+            for path in mpaths:
+                path_stat = multipath.get_path_status(path)
+                if path_stat[0] == 'failed' or path_stat[2] == 'faulty':
+                    count += 1
+        self.log.info("total mpaths = %s", total_mpaths)
+        self.log.info("number of ports disables = %s", len(test_ports))
+        self.log.info("Total Number of ports = %s", len(self.port_ids))
+        self.log.info("count value, paths faild = %s", count)
+        if count != total_mpaths * len(test_ports) / len(self.port_ids):
+            self.error("all paths did not failed on port disable")
+
+    def mpath_pass_check(self):
+        '''
+        checking mpath disk status after enabling the switch port
+        '''
+        wwids = multipath.get_multipath_wwids()
+        err_paths = []
+        for wwid in wwids:
+            if wwid not in process.system_output('multipath -ll',
+                                                 ignore_status=True,
+                                                 shell=True):
+                continue
+            mpaths = multipath.get_paths(wwid)
+            for path in mpaths:
+                path_stat = multipath.get_path_status(path)
+                if path_stat[0] == 'failed' or path_stat[2] == 'faulty':
```

Comment:
Why checking 'failed' and 'faulty' in both functions ?

Reply:
each has some different usage in the if. So this is also needed  in both the places.

Reply:
Need to check the same thing in both `mpath_pass_check()` and `mpath_fail_check()` ? Please clarify